pyproject.toml

```diff
@@ -42,6 +42,8 @@ parse-type = "*"
 pytest = ">=6.2.0"
 typing-extensions = "*"
 packaging = "*"
+gherkin-official = "^29.0.0"
+pydantic = "^2.9.0"
```

Comment:
Id really prefer if we can avoid using pydantic. Its design mixes up model with represantion layer, which is not something I like.

Plus it uses the word `validate` where it should actually say `deserialize`/`load`/`unmarshal`/… — I find it a poorly design API

Reply:
It’d be great if we can just use data classes, or if that’s not enough, `attrs`.

Reply:
Re-implemented with attrs and factory methods for initialise the dataclasses from dicts

Reply:
Apologies - I misread your comment and thought you said "but that's not enough" and so used attrs.  I think dataclasses are fine.  I will commit with removing attr and just using pure dataclasses